### PR TITLE
Support for WebSockets URL resolution

### PR DESCRIPTION
**Pull request by gitarthasarma**
**Support for WebSocket `URL` resolution**
Features added:  
1. First it detects whether the URL address entered is correct, if not valid then point out what is wrong, for example, invalid port number, invalid protocol(ws).
2. After that it checks whether the server is running on the IP and port provided. If yes, then attempts to connect and proceeds, else exits.  

**Issue no:** #68

### CODE REVIEW

**Comment by t0xic0der** [comment]
Let me take a look at this. :smile:

**Comment by t0xic0der** [comment]
@nat236919 and @Vivek-blip, while I am looking into the functionality - could you folks please review if at all any commit has been undone or gone sideways with this push or not? Thanks in advance.

File: cnew.py
Comment (t0xic0der) [inline cnew.py:130]: That's an intelligent way of splitting the URI into protocol, address and port number, facilitating error correction to a greater detail. Great work, @gitarthasarma :star2:!

File: cnew.py
Comment (t0xic0der) [inline cnew.py:147]: The error messages are just as informative as they should be.

**Review by t0xic0der — APPROVED** [review]
Good work, @gitarthasarma. I will wait on @nat236919 and @Vivek-blip to run a check on the code to see inclusions/exclusions. Once I have a green signal from their sides as well, I will initiate a merge.

**Comment by t0xic0der** [comment]
@gitarthasarma while you are waiting for approval, how about you go ahead and :star: this repository if you think this is good enough for people to discover it? :wink:

**Comment by gitarthasarma** [comment]
> @gitarthasarma while you are waiting for approval, how about you go ahead and  this repository if you think this is good enough for people to discover it?

Thanks @t0xic0der I liked the project a lot. Hope to contribute more in this repository. :smile:

**Comment by gitarthasarma** [comment]
I have updated the PR with the current version of the repository. Could you check it? @t0xic0der @nat236919 @Vivek-blip

**Review by nat236919 — APPROVED** [review]
@gitarthasarma @t0xic0der 
According to the file changed, I did not see any exclusions or conflicts on the file, only inclusions.

So, LGTM in terms of being up-to-date.

**Comment by t0xic0der** [comment]
> @gitarthasarma @t0xic0der
> According to the file changed, I did not see any exclusions or conflicts on the file, only inclusions.
> 
> So, LGTM in terms of being up-to-date.

Thanks for taking some time out to review the PR @nat236919.

I am merging this PR.

**Pull request merged**
